sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DeadStores.cs

```diff
@@ -44,9 +44,10 @@ public class DeadStores : SonarDiagnosticAnalyzer
 
         private static readonly DiagnosticDescriptor rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-
         protected sealed override DiagnosticDescriptor Rule => rule;
 
+        private static readonly ISet<string> AllowedValues = new HashSet<string> { "-1", "0", "1" };

```

Comment:
You could rename this to AllowedNumericValues... Or put "null", "true" and "false" inside and lose most of the code in IsAllowedInitialization (the IsKind checks).